.github/workflows/promote.yml

```diff
@@ -0,0 +1,91 @@
+name: promote-contracts
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        required: true
+        type: string
+  workflow_dispatch:
+    inputs:
+      environment:
+        required: true
+        type: choice
+        options:
+          - sbx
+          - prod_test
+          - prod
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  promote:
+    runs-on: self-hosted
+    steps:
+      - name: Set environment
+        id: set-env
+        run: echo "env_name=${{ inputs.environment }}" >> $GITHUB_OUTPUT
+
+      - name: Define source/destination config
+        id: repos
+        run: |
+          case "${{ inputs.environment }}" in
+            test)
+              echo "SOURCE_ENV=test" >> $GITHUB_OUTPUT
+              echo "SOURCE_REPO=ab2d-impl-services" >> $GITHUB_OUTPUT
+              echo "DEST_REPO=ab2d-sbx-services" >> $GITHUB_OUTPUT
+              ;;
+            prod_test)
+              echo "SOURCE_ENV=test" >> $GITHUB_OUTPUT
+              echo "SOURCE_REPO=ab2d-impl-services" >> $GITHUB_OUTPUT
+              echo "DEST_REPO=ab2d-prod-test-services" >> $GITHUB_OUTPUT
+              ;;
+            prod)
+              echo "SOURCE_ENV=test" >> $GITHUB_OUTPUT
+              echo "SOURCE_REPO=ab2d-impl-services" >> $GITHUB_OUTPUT
+              echo "DEST_REPO=ab2d-prod-services" >> $GITHUB_OUTPUT
+              ;;
```

Comment:
SOURCE_ENV and SOURCE_REPO no longer need to be included in the case since they're the same for all.